### DISCUSSION TITLE
Unhandled Runtime Error: Call Stack <unknown>

### DISCUSSION THREAD

**Discussion by ghost** [vercel/next.js#38494] [Help] [answered]
Hi Everyone

When i open the dev tools on Google Chrome, there will be a modal which title is "Unhandled Runtime Error"

I'm sharing the screen shot and you can see the error modal.

Only i have installed Next.js (v12.2.2), React (v18.0.2) with typescript, tailwindcss(v3.1.5) and jest(v28.1.2). 

How can i resolve this problem?

![Ekran Resmi 2022-07-10 23 03 36](https://user-images.githubusercontent.com/78415015/178160438-75039c6a-a129-441c-a422-178a0a757849.png)

**Comment by ghost** [comment]
Anybody help about this problem?

**Reply by icyJoseph** [reply]
Hi,

I saw this the other day... but it happens when I do something "wrong" in the dev tools, like:

```js
let foo = null
foo.bar
```

<img width="976" alt="Screenshot 2022-07-12 at 21 48 36" src="https://user-images.githubusercontent.com/21013447/178581439-0ccd5fc8-c4d2-474d-8f77-a928bec37e2e.png">

I guess, there's some window on error handler that's triggering when something goes wrong on the console. 

I also guess that a 3rd party script could trigger this if they try to do something wrong? I'll do some testing.

**Reply by ghost** [reply]
@icyJoseph 

Thank you for your reply... I'm gonna wait your test. 

After testing, can you send me message about this problem?

**Reply by icyJoseph** [reply]
And this happens only when you open the **console** right? Not as part of normal development. I found that there's indeed a few event listeners for errors, and I guess that's the root of the issue.

**Reply by ghost** [reply]
@icyJoseph 

Yes... Only when i open dev tools, this happens and show this modal.

**Reply by icyJoseph** [reply]
Alright, well, this is sort of expected behaviour mixed with externals.

What I mean is that, Next.js is listening to runtime errors in order to show the Overlay, and not crash out on a blank page, etc.

There must be some kind of script, user code, that's being triggered when you open your console, which in turn causes a runtime error, because of faulty code.

I can think of a `scroll`/`resize` event listener, or perhaps a resize observer? that are running some faulty code. If you resize your browser window, do you see the error as well?

**Reply by ghost** [reply]
@icyJoseph 

Hi again,

I checked listener event especially resizing as you said. But i didn't see any error. For now only I see this error when i open dev tools on Google Chrome. Also I'm using MacOS platform. (macOS Monterey v12.4)

**Reply by icyJoseph** [reply]
Is there a repository, where I could take a look at this? You can even add me as collaborator, and I post back here. I'm genuinely curious, because I saw this happen a while back, but didn't stop to think much of it.

**Reply by ghost** [reply]
@icyJoseph 

I wish I could share it, but it is a company private repo, I can not share it because of license agreement. :( 

But if you want i can share a screenshot of package.json file... So you can create a repo, install packages and test on your computer:

Thank you so much for your support and all messages ! 👍 

<img width="370" alt="Ekran Resmi 2022-07-13 21 57 53" src="https://user-images.githubusercontent.com/78415015/178810582-2283fa99-e17d-46dc-b0a4-ce6b487ed89c.png">

**Reply by icyJoseph** [reply]
Interesting, I was expecting more runtime dependencies. I guess the offending code could be in your components/pages then, but this is really difficult to pinpoint without more info. 

Perhaps if you add your own error listener?

```js
// _app

if(typeof window !== 'undefined') {
  const handler = (event) => {
    console.log(event); // do better logging here
  };
  window.addEventListener('error', handler);
  window.addEventListener('unhandledrejection', handler);
}
```

I think that should catch a whole bunch of errors, including the ones that trigger the dev-overlay, and you can use that to gain more info into where the app is crashing out.

**Reply by ghost** [reply]
@icyJoseph 

> ```js
> // _app
> 
> if(typeof window !== 'undefined') {
>   const handler = (event) => {
>     console.log(event); // do better logging here
>   });
>   window.addEventListener('error', handler);
>   window.addEventListener('unhandledrejection', handler);
> }
> ```
> 


I tried these codes.. After that in console i saw this message: 

<img width="1076" alt="Ekran Resmi 2022-07-13 23 00 33" src="https://user-images.githubusercontent.com/78415015/178824281-4fc5a33b-c799-4fc7-a918-cefe8c799197.png">

Can i contact with you Joseph? Zoom, Google Hangouts, Discord, Slack etc?

PS: Updated comment because of screenshot

**Reply by icyJoseph** [reply]
Sure, you may find me through LinkedIn with the same user handle :)

**Reply by ghost** [reply]
> Sure, you may find me through LinkedIn with the same user handle :)

I sent a invite on LinkedIn for messaging or talk about this problem and also added you as collaborator in this repo. I talked with my company. They said it's ok 😄

**Reply by icyJoseph** [reply]
Alright! I wonder if you have any extensions installed? Does this happen if you open the app in an incognito window? 

I tested just now, and I saw no errors.

**Reply by ghost** [reply]
Yeah ! You're right ! I opened the app in an incognito window. There was a no error when i open Dev Tools on Google Chrome. So there is a problem because of any extension in Chrome. But which one? I have only dev extensions (React, Redux, VueJS etc...) I will check and test all extensions. 

Thank you so much @icyJoseph !! You're best ! Thank you for your all messages !

**Reply by icyJoseph** [reply]
No worries. I'll post an answer. Don't forget to mark as answered, and remove me from the repository! Nice to have more LinkedIn connections though!

**Comment by icyJoseph** [comment, marked as answer]
An extension was causing the error overlay to trigger! That's why it only happened when the console in DevTools was opened.

Because the overlay uses:

```js
  window.addEventListener('error', handler);
  window.addEventListener('unhandledrejection', handler);
```

Errors in the devTools console are also caught. Simply doing `foo.bar` without defining `foo`, triggers it. Of course this all happens in development mode only.

**Reply by ghost** [reply]
I found the extension... If you install VueJS extension on Google Chrome, you see this error modal. Because this repo is not a VueJS Project and when you open dev tools you see a error message like this on console: **"It's not a VueJS Project"** That's why there is an error mode like the screenshot I shared...

@icyJoseph Thank you again for your all support..
